Source/Diagnostics/BTD_Plotfile_Header_Impl.cpp

```diff
@@ -337,3 +343,178 @@ BTDMultiFabHeaderImpl::CopyVec(amrex::Vector<amrex::Real>& dst,
 }
 
 
+BTDSpeciesHeaderImpl::BTDSpeciesHeaderImpl (std::string const & Headerfile_path, std::string const& species_name)
+    : m_Header_path(Headerfile_path), m_species_name(species_name)
+{
+
+}
+
+void
+BTDSpeciesHeaderImpl::ReadHeader ()
+{
+    amrex::Vector<char> HeaderCharPtr;
+    amrex::Long fileLength(0), fileLengthPadded(0);
+    std::ifstream iss;
```

Comment:
You can add stricter exception handling for all `ifstream` variables to catch weird read errors early instead of letting them slip downstream. Helped a lot to find issues with #2335

Usually we set now:
```C++
iss.exceptions(std::ios_base::failbit | std::ios_base::badbit);
```

similar to other `ifstream`s in other files of this PR :)

ref: https://en.cppreference.com/w/cpp/io/basic_ios/exceptions

Reply:
5c8d49f This commit add the exception handling similar to the above mentioned PR and in some other functions within the same file